plutus-metatheory/src/Algorithmic/ReductionEC/Progress.lagda.md

```diff
@@ -82,7 +83,7 @@ progress (unwrap M refl) with progress M
 ... | done (V-wrap V) = step (ruleEC [] (β-wrap V refl) refl refl)
 ... | error E-error = step (ruleErr (unwrap [] / refl) refl)
 progress (con c)      = done (V-con c)
-progress (builtin b / refl) = done (ival b)
+progress {A} (builtin b / refl ) = done (irrValue (ival b btype-∅) )
```

Comment:
is `A` needed?

Reply:
also I don't see why irrValue is needed here, the proof in the argument is `refl`, `ival` wants a a proof of `refl` why doesn't this work without `irrValue`?

Reply:
The `A` is not longer needed. It doesn't work without `irrValue` because the proof returned by `ival` is not `refl`. I'll see if modifying the definitions like mentioned above help...

Reply:
What I did was to move the use of `irrValue` to the `ival` definition, rather than at the `ival` uses. I think it also helps to understand why it's needed.

Reply:
I have a solution now. Please check out the latest commit. What I did is to define a version of `btype` where for the empty context the equation we want holds definitionally. What do you think? Is it too dirty a trick?

Reply:
I'll merge since we can change it later.